internal/connector/internal/api/dbapi/processor.go

```diff
@@ -0,0 +1,143 @@
+package dbapi
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+)
+
+type ProcessorDesiredState string
+type ProcessorStatusPhase string
+
+const (
+	ProcessorUnassigned ProcessorDesiredState = "unassigned"
+	ProcessorReady      ProcessorDesiredState = "ready"
+	ProcessorStopped    ProcessorDesiredState = "stopped"
+	ProcessorDeleted    ProcessorDesiredState = "deleted"
+
+	ProcessorStatusPhaseAssigning      ProcessorStatusPhase = "assigning"      // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseAssigned       ProcessorStatusPhase = "assigned"       // set by kas-fleet-manager - worker
+	ProcessorStatusPhaseUpdating       ProcessorStatusPhase = "updating"       // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseStopped        ProcessorStatusPhase = "stopped"        // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseProvisioning   ProcessorStatusPhase = "provisioning"   // set by kas-agent
+	ProcessorStatusPhaseReady          ProcessorStatusPhase = "ready"          // set by the agent
+	ProcessorStatusPhaseFailed         ProcessorStatusPhase = "failed"         // set by the agent
+	ProcessorStatusPhaseDeprovisioning ProcessorStatusPhase = "deprovisioning" // set by kas-agent
+	ProcessorStatusPhaseDeleting       ProcessorStatusPhase = "deleting"       // set by the kas-fleet-manager - user request
+	ProcessorStatusPhaseDeleted        ProcessorStatusPhase = "deleted"        // set by the agent
+)
+
+var ValidProcessorDesiredStates = []string{
+	string(ProcessorUnassigned),
+	string(ProcessorReady),
+	string(ProcessorStopped),
+	string(ProcessorDeleted),
+}
+
+var AgentProcessorStatusPhase = []string{
+	string(ProcessorStatusPhaseProvisioning),
+	string(ProcessorStatusPhaseDeprovisioning),
+	string(ProcessorStatusPhaseStopped),
+	string(ProcessorStatusPhaseReady),
+	string(ProcessorStatusPhaseFailed),
+	string(ProcessorStatusPhaseDeleted),
+}
+
+type Processor struct {
```

Comment:
I didn't add `channel` to `Processor`'s as I assume it was `connector` specific.

Reply:
the meaning of channels as I have understood them is in the context of "software distribution channels" so a way to manage different level of maturity of a connector for example. I think might be useful for processors too?

I admit that we have mostly used stable as channel for now, so might be worth revisiting this.

Reply:
I see the usefulness of channel for the different connectors as there are a myriad of different ones, some of which may be stable or experimental etc. However there is only one type of processor and the configuration of which determines its behaviour.

That said I'd like to follow RHOCs approach to problems or requirements you've no doubt experienced in the past.

So if you think Processors should have channel too I am happy to add back. At this point I am trying flow your best practices.

Reply:
I'm working on a doc about how channels and other things should work and enhanced for connectors so the definition of channels is not 100% clear however, the original intent of channels were also to provide a way to give users a chance to test new feature which should apply seamless a also to processors (i.e. one day we may want to introduce decisions tables kameletm, etc)